### DISCUSSION TITLE
Add "Release notes" to Extensions

### DISCUSSION THREAD

**Discussion by tristanbob** [4ian/GDevelop#3566] [Enhancement ideas (and their technical discussions)]
## Description

Extension creators will occasionally make versions that include breaking changes which existing users need to review and possibly take action.

Users need a quick way to see this information, without adding a lot of text to the extension description.

## Solution suggested

- [ ] Add a new text area to extensions for extension creators to add "Release notes"
- [ ] Add a way to display these "Release notes" when viewing an extension (but I don't think it should be visible by default).  Perhaps add a "View Release Notes" hyperlink to the right of the version number.
- [ ] When upgrading an extension, the prompt should show the "Installed version" and the "Proposed version" and a way to view the "Release notes"

## Examples of Release notes
![Screenshot 2022-01-29 193545](https://user-images.githubusercontent.com/8879811/151684556-9f9046da-2795-4b09-9635-17b8dd40a477.png)

![Screenshot 2022-01-29 193901](https://user-images.githubusercontent.com/8879811/151684613-a27811d1-35c0-4b15-8793-51ae5444c8ae.png)

**Comment by arthuro555** [comment]
Semi related, a feature I have been thinking of would be "Migrations": We currently have the problem of wanting to rename the functions of older extensions, often made before the best practices and reviews were clearly defined, but cannot easily do so as it would break users games. As a way to mitigate this, GDevelop could propose when renaming something inside of an extension to add it to a list of "Migrations", and when installing the extension update, GDevelop would run those migrations to automatically rename older functions to use the new name.

**Reply by 4ian** [reply]
That's an interesting idea! The editor could tell about it when updating an extension with migrations ("your project will be updated and should continue to work, but be sure to double check everything"), with no migrations ("there should not be any changes to do") or with a migration marked "manual" ("you need to update your project manually for this: blablabla").
